### ISSUE TITLE
Invalid signatures don't raise errors

### ISSUE THREAD

**Issue by KAAtheWiseGit** [nushell/nushell#11266]
### Describe the bug

When making a custom command with an invalid pipe signature, an error won't be raised.
### How to reproduce

```Nushell
~> def command [] <table> -> <string> {}
~> help command
Usage:
  > command 

Flags:
  -h, --help - Display the help message for this command

Input/output types:
  ╭───┬───────┬────────╮
  │ # │ input │ output │
  ├───┼───────┼────────┤
  │ 0 │ any   │ any    │
  ╰───┴───────┴────────╯
```

### Expected behavior

A syntax error.

### Configuration

| key                | value                                                        |
| ------------------ | ------------------------------------------------------------ |
| version            | 0.87.1                                                       |
| branch             |                                                              |
| commit_hash        |                                                              |
| build_os           | linux-x86_64                                                 |
| build_target       | x86_64-alpine-linux-musl                                     |
| rust_version       | rustc 1.72.1 (d5c2e9c34 2023-09-13) (Alpine Linux 1.72.1-r0) |
| cargo_version      | cargo 1.72.1                                                 |
| build_time         | 2023-11-27 13:49:51 +00:00                                   |
| build_rust_channel | release                                                      |
| allocator          | mimalloc                                                     |
| features           | default, sqlite, trash, which, zip                           |
| installed_plugins  |                                                              |

**Comment by kubouch** [comment]
I believe all these issues originate from this problem:
* https://github.com/nushell/nushell/issues/12893
* https://github.com/nushell/nushell/issues/13736
* https://github.com/nushell/nushell/issues/13748
* https://github.com/nushell/nushell/issues/13926
* https://github.com/nushell/nushell/issues/14170

**Comment by texastoland** [comment]
I hope to revive my PR soon 🙏🏼

**Comment by WindSoilder** [comment]
Closed by #14510
